abi/bind/base.go

```diff
@@ -100,22 +100,33 @@ func NewBoundContract(address common.Address, abi abi.ABI, caller ContractCaller
 // DeployContract deploys a contract onto the Ethereum blockchain and binds the
 // deployment address with a Go wrapper.
 func DeployContract(opts *TransactOpts, abi abi.ABI, bytecode []byte, backend ContractBackend, params ...interface{}) (common.Address, *types.Transaction, *BoundContract, error) {
+	// Otherwise try to deploy the contract
+	addr, tx, _, c, err := deploy(opts, abi, bytecode, backend, params...)
+	return addr, tx, c, err
+}
+func DeployContractGetReceipt(opts *TransactOpts, abi abi.ABI, bytecode []byte, backend ContractBackend, params ...interface{}) (*types.Transaction, *types.Receipt, *BoundContract, error) {
+	// Otherwise try to deploy the contract
+	_, tx, receipt, c, err := deploy(opts, abi, bytecode, backend, params...)
+	return tx, receipt, c, err
+}
+
+func deploy(opts *TransactOpts, abi abi.ABI, bytecode []byte, backend ContractBackend, params ...interface{}) (common.Address, *types.Transaction, *types.Receipt, *BoundContract, error) {
 	// Otherwise try to deploy the contract
 	c := NewBoundContract(common.Address{}, abi, backend, backend, backend)
 
 	input, err := c.abi.Pack("", params...)
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, nil, nil, err
 	}
 	tx, receipt, err := c.transact(opts, nil, append(bytecode, input...))
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, nil, nil, err
 	}
 	if receipt == nil {
-		return common.Address{}, nil, nil, errors.New("deploy failed, receipt is nil")
+		return common.Address{}, nil, nil, nil, errors.New("deploy failed, receipt is nil")
 	}
 	c.address = receipt.ContractAddress
-	return c.address, tx, c, nil
+	return c.address, tx, receipt, c, nil
```

Comment:
```suggestion
	return tx, receipt, c, nil
```
let `DeployContract ` get contract address from receipt